### PR TITLE
Add issue templates

### PR DESCRIPTION
**Pull request by microDev1**
- Adds templates for  `Bug Report` and `Feature Request`.
- Also adds links to forum and discord server.

issue_page_ui

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
+name: Bug report
+about: Create a report to help us improve
+title: ''
+labels: ''
```
Comment (tannewt) [inline .github/ISSUE_TEMPLATE/bug_report.md:5]:
```suggestion
labels: 'bug'
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+contact_links:
+  - name: Adafruit Forum
+    url: https://forums.adafruit.com/
+    about: Not sure if a bug is indeed a bug... post it here.
+  - name: Adafruit Discord
+    url: https://adafru.it/discord
+    about: Similar to the forum but more exposure and prompt replies.
```
Comment (tannewt) [inline .github/ISSUE_TEMPLATE/config.yml:7]:
```suggestion
    about: Unofficial chat with many helpful folks and normally prompt replies.
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+contact_links:
+  - name: Adafruit Forum
+    url: https://forums.adafruit.com/
+    about: Not sure if a bug is indeed a bug... post it here.
```
Comment (tannewt) [inline .github/ISSUE_TEMPLATE/config.yml:4]:
```suggestion
    about: Official Adafruit technical support forum. Good for getting help on getting a project working.
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+---
+name: Feature request
+about: Suggest an idea for this project
+title: ''
+labels: ''
```
Comment (tannewt) [inline .github/ISSUE_TEMPLATE/feature_request.md:5]:
```suggestion
labels: 'enhancement'
```

**Review by tannewt — CHANGES_REQUESTED** [review]
Thanks for the PR! This does look very nice! It's cool that we can link out to the forum and chat.

**Comment by ismael-miguel** [comment]
As per recommendation in https://github.com/adafruit/circuitpython/issues/4291#issuecomment-789852208, I'm giving my (edited) feedback here:

In my opinion, to the error template, it would be great to add an optional "Expected behaviour" section.

Some issues, like #4311, the expected behaviour is to do not trigger adjacent pins. (Duh!)
Others, like my issue #4279, the expected behaviour is that it should throw an exception because the filesystem is readonly.
Others, like #4300, could be that the frequency should be different or something else.

This could be added to the "Addicional information", but it would probably be drowned out by other things, and possibly missed.

**Comment by microDev1** [comment]
> it would be great to add an optional "Expected behavior" section.

Thanks! for the feedback @ismael-miguel. I expect the _"Expected behavior"_ to be part of the _Behavior_ section. Its not necessary to adhere to the format provided. The template just provides a minimal structure that can be further modified by the issue filler's needs.

**Comment by microDev1** [comment]
> it would be great to add an optional "Expected behavior" section.

Thanks! for the feedback @ismael-miguel. I expect the _"Expected behavior"_ to be part of the _Behavior_ section. Its not necessary to adhere to the format provided. The template just provides a minimal structure that can be further modified by the issue filler.

**Comment by ismael-miguel** [comment]
Thank you for your insight @microDev1.

Personally, I would assume that *Behaviour* is meant as *Observed Behaviour*, therefore the suggestion to add the *Excepted Behaviour* section, to keep the waters separated.

But, in the end, I agree that the user can just ignore the template, despite being highly recommended to follow it.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+about: Create a report to help us improve
+title: ''
+labels: 'bug'
+assignees: ''
+
+---
+
+Thanks! for testing out CircuitPython. Now that you have got a problem... you can
+file a bug report for it. Feel free to modify the below format to better suit your issue.
+Also, don't forget to remove me... :)
```
Comment (ajs256) [inline .github/ISSUE_TEMPLATE/bug_report.md:12]:
```suggestion
<!-- Thanks! for testing out CircuitPython. Now that you have got a problem... you can
file a bug report for it. Feel free to modify the below format to better suit your issue. -->
```

HTML comments work in GitHub Markdown. Comments are usually used for instructions like this.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+**Firmware**
+
+```python
+Adafruit CircuitPython 6.2.0-beta.2 on 2021-03-01; Raspberry Pi Pico with rp2040
+```
+**Code/REPL**
+
```
Comment (ajs256) [inline .github/ISSUE_TEMPLATE/bug_report.md:20]:
```suggestion
<!-- Include your code that reproduces the bug here. Try to distill down to the minimum possible to reproduce. -->
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+assignees: ''
+
+---
+
+Thanks! for testing out CircuitPython. Now that you have got a problem... you can
+file a bug report for it. Feel free to modify the below format to better suit your issue.
+Also, don't forget to remove me... :)
+
+**Firmware**
+
```
Comment (ajs256) [inline .github/ISSUE_TEMPLATE/bug_report.md:15]:
```suggestion
<!-- Include the version of CircuitPython you're running. You can see it in the `boot_out.txt` file, as well as in the REPL. -->
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+Adafruit CircuitPython 6.2.0-beta.2 on 2021-03-01; Raspberry Pi Pico with rp2040
+```
+**Code/REPL**
+
+```python
+import busio, bitbangio
+i2c = bitbangio.I2C(board.GP1, board.GP0)
+```
+**Behavior**
+
```
Comment (ajs256) [inline .github/ISSUE_TEMPLATE/bug_report.md:26]:
```suggestion
<!-- What happens when you run the code above? Include any error messages. -->
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+Traceback (most recent call last):
+  File "<stdin>", line 1, in <module>
+TimeoutError: Clock stretch too long
+```
+**Description** _(optional)_
+
+- Error while using i2c...
+- Only happens when...
+- might be related to #4291...
+
```
Comment (ajs256) [inline .github/ISSUE_TEMPLATE/bug_report.md:37]:
```suggestion
**Description**
<!-- Optionally, describe the issue in more detail. Here are some examples:
- Error while using i2c...
- Only happens when...
- might be related to #4291...
-->
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+**Additional Info** _(optional)_
+
+Any other information like hardware connection, scope output etc.
```
Comment (ajs256) [inline .github/ISSUE_TEMPLATE/bug_report.md:40]:
```suggestion
**Additional Info**
<!-- Optionally, add any other info, like hardware connection, oscilloscope output etc. -->
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 11]
```diff
@@ -0,0 +1,7 @@
+contact_links:
+  - name: Adafruit Forum
+    url: https://forums.adafruit.com/
+    about: Official Adafruit technical support forum. Good for getting help on getting a project working.
+  - name: Adafruit Discord
+    url: https://adafru.it/discord
+    about: Unofficial chat with many helpful folks and normally prompt replies.
```
Comment (ajs256) [inline .github/ISSUE_TEMPLATE/config.yml:7]:
```suggestion
    about: Official chat group, with many helpful folks and normally prompt replies.
```
I believe the Discord is official.

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+name: Feature request
+about: Suggest an idea for this project
+title: ''
+labels: 'enhancement'
+assignees: ''
+
+---
+
+We are always adding new features and enhancements to CircuitPython 🚀
+and would love ❤ to see what new challenge you have got for us... 🙂
```
Comment (ajs256) [inline .github/ISSUE_TEMPLATE/feature_request.md:11]:
```suggestion
<!--
We are always adding new features and enhancements to CircuitPython 🚀
and would love ❤ to see what new challenge you have got for us... 🙂
-->
```

**Review by ajs256 — COMMENTED** [review]
Looks great! I just suggest some changes to add clearer instructions.

File: .github/ISSUE_TEMPLATE/config.yml (hunk 11 above)
Comment (microDev1) [reply .github/ISSUE_TEMPLATE/config.yml:7]: This was suggested by @tannewt [above](https://github.com/adafruit/circuitpython/pull/4316#discussion_r586593147).

Comment (ajs256) [reply .github/ISSUE_TEMPLATE/config.yml:7]: Ooh, didn't see that.

**Comment by microDev1** [comment]
@ajs256 Thanks! for the suggestions. I am not adding all of the tags as they cause the enclosed content to not show up in preview.

**Review by tannewt — APPROVED** [review]
Great! Let's start with this. Thank you!

**Pull request merged**